### ISSUE TITLE
Opengl 3d rendering on AMD is broken for Pokemon Black Version 2 when using dedicated GPU on laptop

### ISSUE THREAD

**Issue by awsomekevin12** [melonDS-emu/melonDS#1688]
System Specs: 
Laptop: Rog Strix Advantage Edition 2021, G513QY
GPU: RX 6800M
iGPU: RX Vega 8 5000
AMD Adrenalin version: 23.4.3
Windows Version: Windows 10 Home 10.0.19045 Build 19045
This is my user benchmark for RAM, SSD, etc: https://www.userbenchmark.com/UserRun/47817255
I think the issue lies with my dedicated GPU.

Normally, the OpenGL renderer causes insane graphical glitches with 3d games on my dedicated GPU (I think this occurred after AMD overhauled their OpenGL drivers). Here's an example of the graphical errors I have in a 3d section of Pokemon Black Version 2 (this is the poke mart):
#1578 

However, the graphical glitches disappear after switching to my integrated graphics. This is what the poke mart looks like running on my iGPU:
![image](https://github.com/melonDS-emu/melonDS/assets/134647457/fc512065-0aa3-4b8d-9984-c46d5db2ce5f)
OpenGL works perfectly in this case.

**Comment by RSDuck** [comment]
probably just different quirks of the driver/GPU architecture ¯\_(ツ)_/¯

**Comment by taufik-ridwan** [comment]
Yep, i can confirm the broken rendering only affects AMD gpu, and it works just fine in nvidia/intel. Thankfully, The new OpenGL (Compute shader) from the `compute-shader-renderer` branch fixes the issue on AMD gpu (tested with AW:Dual Strike)

**Comment by Woffingshire** [comment]
> Yep, i can confirm the broken rendering only affects AMD gpu, and it works just fine in nvidia/intel. Thankfully, The new OpenGL (Compute shader) from the `compute-shader-renderer` branch fixes the issue on AMD gpu (tested with AW:Dual Strike)

Where would one go about obtaining this new OpenGL computer shader?

**Comment by WalkerBait39** [comment]
Did anyone manage to fix this issue

**Comment by glenntaylor0** [comment]
> Where would one go about obtaining this new OpenGL computer shader?

You have to build it yourself, follow the build instructions for melonDS in the [readme](https://github.com/melonDS-emu/melonDS/tree/compute-shader-renderer#readme)

To use the compute-shader-renderer branch, when you get to the part where you do:
`git clone https://github.com/melonDS-emu/melonDS`
`cd melonDS`

make sure you do instead
`git clone https://github.com/melonDS-emu/melonDS`
`cd melonDS`
`git checkout compute-shader-renderer`

I just did this in Windows, it was very easy and the instructions were super clear, but I am very adept at this sort of thing so I did notice when I had to re-run some of the update/install commands because they didn't complete correctly the first time. If you get stuck let me know, I'll see if I can send you the melonDS exe I compiled

Once built, it gave a new openGL option which fixed up all my graphical problems with my AMD 6700XT
![image](https://github.com/melonDS-emu/melonDS/assets/29875072/bbcfba48-8970-4bc6-bdc1-e43b9554f683)
![image](https://github.com/melonDS-emu/melonDS/assets/29875072/cefd0e19-405d-4341-b42e-550c24edfcff)
![image](https://github.com/melonDS-emu/melonDS/assets/29875072/669a06c7-249f-4689-be2b-8b5c5a0166bd)
![image](https://github.com/melonDS-emu/melonDS/assets/29875072/8c34cae8-a3cc-4c37-985c-5d5d7ec4c2bf)

**Comment by Zer077** [comment]
> `git checkout compute-shader-renderer`

can you share the melonds exe compiled? i tried to buil this version but was imposible
